include/deal.II/base/aligned_vector.h

```diff
@@ -1553,6 +1590,51 @@ AlignedVector<T>::insert_back(ForwardIterator begin, ForwardIterator end)
 
 
 
+template <class T>
+template <typename RandomAccessIterator, typename>
+inline typename AlignedVector<T>::iterator
+AlignedVector<T>::insert(const_iterator       position,
+                         RandomAccessIterator begin,
+                         RandomAccessIterator end)
+{
+  Assert(this->begin() <= position && position <= this->end(),
+         ExcMessage("The position iterator is not valid."));
+  const auto offset = position - this->begin();
+
+  const size_type old_size   = size();
+  const size_type range_size = end - begin;
+  const size_type new_size   = old_size + range_size;
+  if (range_size != 0)
+    {
+      // This is similar to allocate_and_move(), except that we need to move
+      // whatever was before position and whatever is after it into two
+      // different places
+      T *new_data_ptr = nullptr;
+      Utilities::System::posix_memalign(
+        reinterpret_cast<void **>(&new_data_ptr), 64, new_size * sizeof(T));
+
+      // Correctly handle the case where the range is inside the present array
+      // by creating a temporary.
+      AlignedVector<T> temporary(begin, end);
+      dealii::internal::AlignedVectorMoveConstruct<T *, T>(
+        elements.get(), elements.get() + offset, new_data_ptr);
+      dealii::internal::AlignedVectorMoveConstruct<T *, T>(
+        temporary.begin(), temporary.end(), new_data_ptr + offset);
+      dealii::internal::AlignedVectorMoveConstruct<T *, T>(
+        elements.get() + offset,
+        elements.get() + old_size,
+        new_data_ptr + offset + range_size);
+
+      Deleter deleter(this);
+      elements          = decltype(elements)(new_data_ptr, std::move(deleter));
+      used_elements_end = elements.get() + new_size;
+      allocated_elements_end = elements.get() + new_size;
+    }
+  return this->begin() + offset;
+}
```

Comment:
Is this going to work if the data was replicated across processes? I suspect not, right? You may want to prohibit that with an assertion.

Reply:
No, it definitely will not work in that case since we are not using `MPI_Win` to reallocate.

We aren't that consistent in checking these - I'll write a follow-up patch to fix the other places.